File: backend/src/main/java/com/festago/festival/dto/FestivalResponse.java
package com.festago.festival.dto;

import com.festago.festival.domain.Festival;
import java.time.LocalDate;

public record FestivalResponse(
    Long id,
    Long schoolId,
    String name,
    LocalDate startDate,
    LocalDate endDate,
    String thumbnail) {

    public static FestivalResponse from(Festival festival) {
        return new FestivalResponse(
            festival.getId(),
            festival.getSchool().getId(),
            festival.getName(),
            festival.getStartDate(),
            festival.getEndDate(),
            festival.getThumbnail());
    }
}

File: android/festago/data-legacy/src/main/java/com/festago/festago/data/dto/FestivalResponse.kt
package com.festago.festago.data.dto

import com.festago.festago.model.Festival
import kotlinx.serialization.Serializable
import java.time.LocalDate

@Serializable
data class FestivalResponse(
    val id: Int,
    val schoolId: Int,
    val name: String,
    val startDate: String,
    val endDate: String,
    val thumbnail: String,
) {
    fun toDomain(): Festival = Festival(
        id = id.toLong(),
        name = name,
        startDate = LocalDate.parse(startDate),
        endDate = LocalDate.parse(endDate),
        thumbnail = thumbnail,
    )
}
